src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -90,10 +84,15 @@ export class DefaultTelemetryService implements TelemetryService {
         recordSessionEnd({ value: currTime.getTime() - this.startTime.getTime() })
 
         // only write events to disk if telemetry is enabled at shutdown time
+        // clears cache if cache is abnormally large:
+        // call stack while processing cache can overflow if over 25 MB. Let's not let it get that way
         if (this.telemetryEnabled) {
-            try {
-                await writeFile(this.persistFilePath, JSON.stringify(this._eventQueue))
-            } catch {}
+            if (this._eventQueue.length < 1000) {
+                try {
+                    await writeFile(this.persistFilePath, JSON.stringify(this._eventQueue))
+                } catch {}
+            }
+            // TODO: else emit a "failed telemetry cache" metric?
```

Comment:
Suggestion: log something and maybe try to save something in global state (static size, nothing that can grow unbounded). I'd at least want to know if the cache writing failed in the previous session. This is best-effort of course but leaving a trail would be helpful for debugging.